packages/contracts/test/core/permission/permission-manager.ts

```diff
@@ -319,93 +338,126 @@ describe.only('Core: PermissionManager', function () {
 
       await expect(
-        pm.removeOwner(pm.address, ADMIN_PERMISSION_ID, REVOKE_OWNER_FLAG)
+        pm.removeOwner(someWhere, ADMIN_PERMISSION_ID, REVOKE_OWNER_FLAG)
       )
         .to.emit(pm, 'OwnerRemoved')
-        .withArgs(pm.address, ADMIN_PERMISSION_ID, ownerSigner.address, 0);
+        .withArgs(someWhere, ADMIN_PERMISSION_ID, ownerSigner.address, 0, 0);
 
       expect(
-        await pm.getFlags(pm.address, ADMIN_PERMISSION_ID, ownerSigner.address)
+        await pm.getFlags(someWhere, ADMIN_PERMISSION_ID, ownerSigner.address)
       ).to.deep.equal([0, 0]);
     });
 
     it('should revert if a zero flag is passed', async () => {
       await expect(
-        pm.removeOwner(pm.address, ADMIN_PERMISSION_ID, 0)
+        pm.removeOwner(someWhere, ADMIN_PERMISSION_ID, 0)
       ).to.be.revertedWithCustomError(pm, 'FlagCanNotBeZero');
     });
 
     it("should revert if flags that don't exist are removed", async () => {
-      await pm.removeOwner(pm.address, ADMIN_PERMISSION_ID, GRANT_OWNER_FLAG);
+      await pm.removeOwner(someWhere, ADMIN_PERMISSION_ID, GRANT_OWNER_FLAG);
 
       await expect(
-        pm.removeOwner(pm.address, ADMIN_PERMISSION_ID, GRANT_OWNER_FLAG)
+        pm.removeOwner(someWhere, ADMIN_PERMISSION_ID, GRANT_OWNER_FLAG)
       )
         .to.be.revertedWithCustomError(pm, 'InvalidFlagsForRemovalPassed')
         .withArgs(REVOKE_OWNER_FLAG, GRANT_OWNER_FLAG);
     });
 
     it('should correctly decrease owner counters', async () => {
       expect(
-        await pm.getPermissionData(pm.address, ADMIN_PERMISSION_ID)
+        await pm.getPermissionData(someWhere, ADMIN_PERMISSION_ID)
       ).to.deep.equal([true, 1, 1]);
 
       const newOwner = otherSigner.address;
 
       await pm.addOwner(
-        pm.address,
+        someWhere,
         ADMIN_PERMISSION_ID,
         newOwner,
         FULL_OWNER_FLAG
       );
 
       expect(
-        await pm.getPermissionData(pm.address, ADMIN_PERMISSION_ID)
+        await pm.getPermissionData(someWhere, ADMIN_PERMISSION_ID)
       ).to.deep.equal([true, 2, 2]);
 
       await pm
         .connect(otherSigner)
-        .removeOwner(pm.address, ADMIN_PERMISSION_ID, GRANT_OWNER_FLAG);
+        .removeOwner(someWhere, ADMIN_PERMISSION_ID, GRANT_OWNER_FLAG);
 
       expect(
-        await pm.getPermissionData(pm.address, ADMIN_PERMISSION_ID)
+        await pm.getPermissionData(someWhere, ADMIN_PERMISSION_ID)
       ).to.deep.equal([true, 1, 2]);
 
       await pm
         .connect(otherSigner)
-        .removeOwner(pm.address, ADMIN_PERMISSION_ID, REVOKE_OWNER_FLAG);
+        .removeOwner(someWhere, ADMIN_PERMISSION_ID, REVOKE_OWNER_FLAG);
 
       expect(
-        await pm.getPermissionData(pm.address, ADMIN_PERMISSION_ID)
+        await pm.getPermissionData(someWhere, ADMIN_PERMISSION_ID)
       ).to.deep.equal([true, 1, 1]);
     });
+
+    it('should also remove delegatee flags when removing the owner flags', async () => {
```

Comment:
I would also add a test tp check that, when granting or revoking if the permission owners has also delegation the delegation is not removed

Reply:
Added in https://github.com/aragon/osx/pull/611/commits/6abd9fce37ed912bae16184fa33409fe2403fe7b